### ISSUE TITLE
Links that share domain with metabase result in incorrect URLs

### ISSUE THREAD

**Issue by mbosecke** [metabase/metabase#47036]
### Describe the bug

If metabase is hosted on a sub-path (related to #30895?), for example "https://www.mydomain.com/metabase", then if you render links in your question that go to a different path on the same domain, for example "https://www.mydomain.com/otherapp", the end-result is an HTTP link that actually takes you to "https://www.mydomain.com/metabase/otherapp" which is incorrect and will likely result in a 404. 

The desired result is that the final HTTP link respects what was written in the raw sql, in this case "https://www.mydomain.com/otherapp". 

Similar to what was described in #25350  except it only applies to links on the same domain (links on a completely different domain will work correctly).

### To Reproduce

1. Set metabase's site URL to a sub-path, like "https://www.mydomain.com/metabase". 
2. Create a raw sql question with the following query:
`   select 'https://www.google.com' as external_url, 'https://www.mydomain.com/otherapp' as shared_domain_url;`
3. Click the "external url" link and see that it works correctly, but click the "shared_domain_url" and see that it was manipulated in an incorrect way.

### Expected behavior

The "shared_domain_url" should take the user to "https://www.mydomain.com/otherapp".

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-08-19",
      "tag": "v0.50.20.6",
      "hash": "dac50d0"
    },
    "settings": {
      "report-timezone": null
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1-LTS",
    "java.vendor": "Red Hat, Inc.",
    "java.vendor.url": "https://www.redhat.com/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1-LTS",
    "os.name": "Linux",
    "os.version": "5.14.0-340.el9.x86_64",
    "user.language": "en",
    "user.timezone": "America/Edmonton"
  }
}
```

### Severity

Broken questions and blocking an upgrade

### Additional context

I recently upgraded from 0.41.5 and this problem didn't used to occur on that old version. Only after upgrading to 0.50.20.6 was the problem identified.

**Comment by paoliniluis** [comment]
We never supported running Metabase on sub paths. This is something that you need to fix via redirection

**Comment by mbosecke** [comment]
I'd understand if that's a scenario you don't support and therefore will close this ticket. However, to add some clarity, this is not a proxying problem, this is a problem with how Metabase is

**Comment by mbosecke** [comment]
I'd understand if this is not a scenario that you support and therefore want to close this ticket. 

However, I will comment on your second point there just to make sure there isn't confusion. This is not a reverse proxy-rule or redirection issue, but a deficiency with how metabase is constructing/manipulating URLs. The SQL query has a working URL hardcoded into it but then metabase manipulates it into an incorrect URL. This bug would have to be overcome by metabase as a prerequisite to supporting sub-paths (if that was something this project desired to support in the future). It's not a user-error but if it's out-of-scope for this project than so be it.

**Comment by mbosecke** [comment]
Theoretically I _could_ resolve this by using a redirection rule, but that rule would be to effectively undo what metabase is doing with it's url transformation. It would be a workaround, not a great permanent solution.

**Comment by mbosecke** [comment]
If anyone else is hitting this bug you can workaround it by making your links differ from your metabase "Site URL" just slightly by either changing the protocol or the subdomain (whichever might work for your setup). The metabase code will no longer detect that it matches the "Site URL" and will stop mangling the URL and leave it as-is. For example:

If metabase site URL is `https://www.mydomain.com/metabase` and you want to create a link to `https://www.mydomain.com/otherapp` then either of the following will bypass the metabase url mangling code:

- `http://www.mydomain.com/otherapp` (differs by protocol)
- `https://mydomain.com/otherapp` (differs by subdomain)
